src/actions/posts.js

```diff
@@ -30,6 +30,66 @@ import {
 } from './preferences';
 import {getProfilesByIds, getProfilesByUsernames, getStatusesByIds} from './users';
 
+export function receivedPost(post) {
+    return {
+        type: PostTypes.RECEIVED_POST,
+        data: post,
+    };
+}
+
+export function receivedNewPost(post) {
+    return {
+        type: PostTypes.RECEIVED_NEW_POST,
+        data: post,
+    };
+}
+
+export function receivedPosts(posts) {
+    return {
+        type: PostTypes.RECEIVED_POSTS,
+        data: posts,
+    };
+}
+
+export function receivedPostsSince(posts, channelId, since) {
+    return {
+        type: PostTypes.RECEIVED_POSTS_SINCE,
+        channelId,
+        data: posts,
+        since,
+    };
+}
+
+export function receivedPostsInChannel(posts, channelId) {
+    return {
+        type: PostTypes.RECEIVED_POSTS_IN_CHANNEL,
+        channelId,
+        data: posts,
+    };
+}
+
+export function receivedPostsInThread(posts, rootId) {
+    return {
+        type: PostTypes.RECEIVED_POSTS_IN_THREAD,
+        data: posts,
+        rootId,
+    };
+}
+
+export function postDeleted(post) {
+    return {
+        type: PostTypes.POST_DELETED,
+        data: post,
+    };
+}
+
+export function postRemoved(post) {
+    return {
+        type: PostTypes.POST_REMOVED,
+        data: post,
+    };
+}
+
```

Comment:
Good call making the above dispatches into separate functions 👍

Reply:
This would require us to make changes to webapp as well as we are using them `PostTypes.REMOVE_POST` action type

Reply:
Yeah, I'm liking this idea. It also makes it easier to see what the arguments should be without having to find the reducers